azurerm/internal/services/storage/storage_management_policy_resource.go

```diff
@@ -243,172 +311,334 @@ func resourceStorageManagementPolicyDelete(d *schema.ResourceData, meta interfac
+	}
+}
+
+func expandStorageManagementPolicyActionsBaseBlob(inputs []interface{}) *storage.ManagementPolicyBaseBlob {
+	if len(inputs) == 0 {
+		return nil
+	}
+	input := inputs[0].(map[string]interface{})
+	result := storage.ManagementPolicyBaseBlob{}
+
+	if v, ok := input["delete_after_days_since_modification_greater_than"].(float64); ok && v != -1 {
+		result.Delete = &storage.DateAfterModification{
+			DaysAfterModificationGreaterThan: utils.Float(v),
 		}
 	}
-	if _, ok := d.GetOk(fmt.Sprintf("rule.%d.actions", ruleIndex)); ok {
-		if _, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.base_blob", ruleIndex)); ok {
-			baseBlob := &storage.ManagementPolicyBaseBlob{}
-			if v, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.base_blob.0.tier_to_cool_after_days_since_modification_greater_than", ruleIndex)); ok {
-				if v != nil {
-					baseBlob.TierToCool = &storage.DateAfterModification{
-						DaysAfterModificationGreaterThan: utils.Float(float64(v.(int))),
-					}
-				}
-			}
-			if v, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.base_blob.0.tier_to_archive_after_days_since_modification_greater_than", ruleIndex)); ok {
-				if v != nil {
-					baseBlob.TierToArchive = &storage.DateAfterModification{
-						DaysAfterModificationGreaterThan: utils.Float(float64(v.(int))),
-					}
-				}
-			}
-			if v, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.base_blob.0.delete_after_days_since_modification_greater_than", ruleIndex)); ok {
-				if v != nil {
-					baseBlob.Delete = &storage.DateAfterModification{
-						DaysAfterModificationGreaterThan: utils.Float(float64(v.(int))),
-					}
-				}
-			}
-			definition.Actions.BaseBlob = baseBlob
+	if v, ok := input["tier_to_archive_after_days_since_modification_greater_than"].(float64); ok && v != -1 {
+		result.TierToArchive = &storage.DateAfterModification{
+			DaysAfterModificationGreaterThan: utils.Float(v),
+		}
+	}
+	if v, ok := input["tier_to_cool_after_days_since_modification_greater_than"].(float64); ok && v != -1 {
+		result.TierToCool = &storage.DateAfterModification{
+			DaysAfterModificationGreaterThan: utils.Float(v),
 		}
+	}
 
-		if _, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.snapshot", ruleIndex)); ok {
-			snapshot := &storage.ManagementPolicySnapShot{}
-			if v, ok := d.GetOk(fmt.Sprintf("rule.%d.actions.0.snapshot.0.delete_after_days_since_creation_greater_than", ruleIndex)); ok {
-				v2 := float64(v.(int))
-				snapshot.Delete = &storage.DateAfterCreation{DaysAfterCreationGreaterThan: &v2}
-			}
-			definition.Actions.Snapshot = snapshot
+	return &result
+}
+
+func expandStorageManagementPolicyActionsSnapshot(inputs []interface{}) *storage.ManagementPolicySnapShot {
+	if len(inputs) == 0 {
+		return nil
+	}
+	input := inputs[0].(map[string]interface{})
+
+	result := storage.ManagementPolicySnapShot{}
+
+	if v, ok := input["delete_after_days_since_creation_greater_than"].(float64); ok && v != -1 {
```

Comment:
this means this value can't be removed, is that intentional?

Reply:
same above?